### PR TITLE
Upgrade ddtrace

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

ddtrace v0.53.0 is now released : https://github.com/DataDog/dd-trace-py/releases/tag/v0.53.0
which contains 
- https://github.com/DataDog/dd-trace-py/pull/2802
- https://github.com/DataDog/dd-trace-py/pull/2803

### Motivation

Cold start improvements + ever greening

### Testing Guidelines

Unit test + integration tests

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [x] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: tests/integration/snapshots/logs/async-metrics_python27.log [hunk 1]
```diff
@@ -4,7 +4,7 @@ START RequestId: XXXX Version: $LATEST
...
... "function_arn": "arn:aws:lambda:sa-east-1:601427279990:function:integration-tests-python-XXXX-async-metrics_python27", "dd_trace": "X.X.X", "_dd.origin": "lambda", "http.status_code": "200", "resource_names": "integration-tests-python-XXXX-async-metrics_python27", "function_trigger.event_source_arn": "arn:aws:apigateway:sa-east-1::/restapis/wt6mne2s9k/stages/test", "function_version": "$LATEST"}, "error": 0, "duration": XXXX, "type": "serverless", "span_id": "XXXX"}, {"resource": "requests.request", "name": "requests.request", "service": "aws.lambda", "start": XXXX, "trace_id": "XXXX", "metrics": {"_dd.measured": 1}, "parent_id": "XXXX", "meta": {"http.status_code": "200", "http.url": "https://httpstat.us/200/", "_dd.origin": "lambda", "http.method": "GET"}, "error": 0, "duration": XXXX, "type": "http", "span_id": "XXXX"}, {"resource": "requests.request", "name": "requests.request", "service": "aws.lambda", "start": XXXX, "trace_id": "XXXX", "metrics": {"_dd.measured": 1}, "parent_id": "XXXX", "meta": {"http.status_code": "400", "http.url": "https://httpstat.us/400/", "_dd.origin": "lambda", "http.method": "GET"}, "error": 0, "duration": XXXX, "type": "http", "span_id": "XXXX"}]]}
```
Comment (maxday) [inline tests/integration/snapshots/logs/async-metrics_python27.log:7]: This is only a change in service. This is likely coming the change in v49 (from the changelog) `requests: spans will no longer inherit the service name from the parent.` (same applied to all integration test snapshots)
Screen Shot 2021-09-14 at 1 27 47 PM

**Review by DarcyRaynerDD — APPROVED** [review]

File: tests/integration/snapshots/logs/async-metrics_python27.log (hunk 1 above)
Comment (DarcyRaynerDD) [reply tests/integration/snapshots/logs/async-metrics_python27.log:7]: @Hesperide Is this an expected change. Could this blow up the service map again?

Comment (DarcyRaynerDD) [reply tests/integration/snapshots/logs/async-metrics_python27.log:7]: Also, is DD_SERVICE still applied to all these spans by default?

Comment (Hesperide) [reply tests/integration/snapshots/logs/async-metrics_python27.log:7]: What do these traces look like in-app?

File: tests/integration/snapshots/logs/async-metrics_python36.log [hunk 2]
```diff
@@ -4,7 +4,7 @@ START RequestId: XXXX Version: $LATEST
...
...on-tests-python-XXXX-async-metrics_python36", "functionname": "integration-tests-python-XXXX-async-metrics_python36", "datadog_lambda": "X.X.X", "dd_trace": "X.X.X", "function_trigger.event_source": "api-gateway", "function_trigger.event_source_arn": "arn:aws:apigateway:sa-east-1::/restapis/wt6mne2s9k/stages/test", "http.method": "GET", "http.status_code": "200"}, "metrics": {"system.pid": XXXX, "_dd.agent_psr": 1.0, "_sampling_priority_v1": 1}, "type": "serverless"}, {"trace_id": "XXXX", "parent_id": "XXXX", "span_id": "XXXX", "service": "aws.lambda", "resource": "requests.request", "name": "requests.request", "error": 0, "start": XXXX, "duration": XXXX, "meta": {"_dd.origin": "lambda", "http.method": "GET", "http.url": "https://httpstat.us/200/", "http.status_code": "200"}, "metrics": {"_dd.measured": 1}, "type": "http"}, {"trace_id": "XXXX", "parent_id": "XXXX", "span_id": "XXXX", "service": "aws.lambda", "resource": "requests.request", "name": "requests.request", "error": 0, "start": XXXX, "duration": XXXX, "meta": {"_dd.origin": "lambda", "http.method": "GET", "http.url": "https://httpstat.us/400/", "http.status_code": "400"}, "metrics": {"_dd.measured": 1}, "type": "http"}]]}
```
Comment (maxday) [inline tests/integration/snapshots/logs/async-metrics_python36.log:7]: The diff is not related to the service name anymore, it's just a reord
Screen Shot 2021-09-16 at 5 47 18 PM
er

Comment (Hesperide) [reply tests/integration/snapshots/logs/async-metrics_python36.log:7]: Great, thanks for the help here Maxime.

**Review by Hesperide — APPROVED** [review]

**Pull request merged**